fendermint/testing/contract-test/tests/run_upgrades.rs

```diff
@@ -0,0 +1,272 @@
+// Copyright 2022-2024 Protocol Labs
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+mod staking;
+
+use anyhow::{Context, Ok};
+use async_trait::async_trait;
+use ethers::types::U256;
+use fendermint_contract_test::Tester;
+use fendermint_rpc::response::decode_fevm_return_data;
+use rand::rngs::StdRng;
+use rand::SeedableRng;
+use std::str::FromStr;
+
+use ethers::contract::abigen;
+use fvm_shared::address::Address;
+use fvm_shared::bigint::Zero;
+use fvm_shared::econ::TokenAmount;
+use fvm_shared::version::NetworkVersion;
+use tendermint_rpc::Client;
+
+use fendermint_crypto::SecretKey;
+use fendermint_vm_actor_interface::eam;
+use fendermint_vm_actor_interface::eam::EthAddress;
+use fendermint_vm_core::Timestamp;
+use fendermint_vm_genesis::{Account, Actor, ActorMeta, Genesis, PermissionMode, SignerAddr};
+use fendermint_vm_interpreter::fvm::store::memory::MemoryBlockstore;
+use fendermint_vm_interpreter::fvm::upgrades::{Upgrade, UpgradeScheduler};
+use fendermint_vm_interpreter::fvm::{bundle::contracts_path, FvmMessageInterpreter};
+use fendermint_vm_message::chain::ChainMessage;
+
+// returns a seeded secret key which is guaranteed to be the same every time
+fn my_secret_key() -> SecretKey {
+    SecretKey::random(&mut StdRng::seed_from_u64(123))
+}
+
+// this test applies a series of upgrades to the state and checks that the upgrades are applied correctly
+#[tokio::test]
+async fn test_applying_upgrades() {
+    use bytes::Bytes;
+    use fendermint_rpc::message::{GasParams, SignedMessageFactory};
+    use lazy_static::lazy_static;
+
+    lazy_static! {
+        /// Default gas params based on the testkit.
+        static ref GAS_PARAMS: GasParams = GasParams {
+            gas_limit: 10_000_000_000,
+            gas_fee_cap: TokenAmount::default(),
+            gas_premium: TokenAmount::default(),
+        };
+    }
+
+    // this is the contract we want to deploy
+    const CONTRACT_HEX: &str = include_str!("../../contracts/SimpleCoin.bin");
+    // generate type safe bindings in rust to this contract
+    abigen!(SimpleCoin, "../contracts/SimpleCoin.abi");
+    // once we deploy this contract, this is the address we expect the contract to be deployed to
+    const CONTRACT_ADDRESS: &str = "f410fnz5jdky3zzcj6pejqkomkggw72pcuvkpihz2rwa";
+    // the amount we want to send to the contract
+    const SEND_BALANCE_AMOUNT: u64 = 1000;
+
+    let mut upgrade_scheduler = UpgradeScheduler::new();
+    upgrade_scheduler
+        .add(
+            Upgrade::new("mychain", 1, Some(1), |state| {
+                println!(
+                    "[Upgrade at height {}] Deploy simple contract",
+                    state.block_height()
+                );
+
+                // create a message for deploying the contract
+                let mut mf =
+                    SignedMessageFactory::new_secp256k1(my_secret_key(), 1, state.chain_id());
```

Comment:
Do we still have to use this or can we just use the `MessageFactory` that doesn't sign now?